spec.md

```diff
@@ -570,13 +559,297 @@ schema being used.
+
+| Header | Type | Description |
+| --- | --- | --- |
+| X-Broker-API-Version* | string | See [API Version Header](#api-version-header). |
+
+\* Headers with an asterisk are REQUIRED.
+
+#### cURL
+
+```
+$ curl http://username:password@service-broker-url/v2/catalog/schemas -H "X-Broker-API-Version: 2.14"
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | MUST be returned upon successful processing of this request. The expected response body is below. |
+
+
+#### Body
+
+The root JSON Schema document is a complete collection of JSON Schemas
+refrenced in the Service Broker's catalog using `$ref`. The body MUST contain
+all JSON Schema object definitions found in the catalog and referenced
+sub-schemas.
+
+| Response Field | Type | Description |
+| --- | --- | --- |
+| $schema* | string | The JSON Schema declaring the version of JSON schema being used. |
+| $id* | string | The top level JSON Schema ID for this document. |
+| definitions* | Dictionary-of-JSON Schema Object | A collection of JSON sub-Schema Objects. |
+
+Please refer to the [JSON Schema](http://json-schema.org/) specification for
+more details on JSON Schema.
+
+\* Fields with an asterisk are REQUIRED.
+
+```
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "$id": "osb_v2:///schemas/",
+  "definitions": {
+    "foo/v1": {
+      "$id": "osb_v2:///fakeservice/v1",
+      "definitions": {
+        "BillingAccount" : {
+          "$id" : "#BillingAccountParameters",
+          "type": "object",
+          "properties": {
+            "billing-account": {
+              "description": "Billing account number used to charge use of shared fake server.",
+              "type": "string"
+            }
+          }
+        },
+        "bindings": {
+          "$id" : "/bindings",
+          "definitions": {
+            "$id" : "#AccountParameters",
+            "type": "object",
+            "properties": {
+              "billing": {
+                "$ref": "osb_v2:///fakeservice/v1#BillingAccountParameters"
+              },
+              "quota-account": {
+                "description": "Quota account number used to charge use of fake bandwidth.",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+#### Route
+
+`GET /v2/catalog/schemas/:component_path`
+
+#### Headers
+
+The following HTTP Headers are defined for this operation:
+
+| Header | Type | Description |
+| --- | --- | --- |
+| X-Broker-API-Version* | string | See [API Version Header](#api-version-header). |
+
+\* Headers with an asterisk are REQUIRED.
+
+#### cURL
+```
+$ curl http://username:password@service-broker-url/v2/catalog/schemas/:component_path -H "X-Broker-API-Version: 2.13"
+```
+
+### Response
+
+| Status Code | Description |
+| --- | --- |
+| 200 OK | MUST be returned upon successful processing of this request. The expected response body is below. |
```

Comment:
Should 304s be allowed? That would allow better controls over cache-ability of schemas.

The response codes seem written from perspective of the broker (broker returns ...), but there may be other responses that the platform may see, i.e. 304 if there is a caching proxy in-between. Would it be worth clarifying, or is it assumed?

Reply:
I think this is an interesting idea, and perhaps it can be a followup because knowing how to send a 304 requires some kind of etag or cache-control header to also be sent and there is nothing like that in the spec today. If this is a requirement, please make an issue and we can follow up with a spec amendment.